src/Native/Unix/System.Native/pal_time.h

```diff
@@ -13,6 +13,14 @@ typedef struct TimeSpec
     int64_t tv_nsec; // nanoseconds
 } TimeSpec;
 
+typedef struct ProcessCpuInformation
+{
+    uint64_t lastRecordedCurrentTime;
+    uint64_t lastRecordedKernelTime;
+    uint64_t lastRecordedUserTime;
```

Comment:
We already gather this information in the Process class, on macOS using proc_pid_rusage and on Linux by reading it out of procfs.  Should we switch to using this API in the PAL instead?  If no, what's the difference?  And if yes, can you do that as part of this change so that we have built-in validation of it via our public surface area (as well as the corresponding Interop.*.cs files)?

Reply:
Ah, I may have answered my own question: getrusage doesn't let you get the data for an arbitrary process by ID.

Reply:
I searched for any prior usage of the `getrusage` API first, but there were not any. Since this is quite performance sensitive API I would expect `getrusage` to be faster for this particular use case than parsing procfs. It can't be used in the other place for the exact reason you stated.